README.md

```diff
@@ -32,6 +32,28 @@ To provide values for these configuration values, they must be either:
 **OR**
 * Passed as compile time preprocessor macros
 
+## Porting the coreMQTT Agent Library
+In order to use the MQTT Agent library on a platform, you need to supply thread safe functions for the agent's [messaging interface](source/include/core_mqtt_agent_message_interface.h).
+
+### Messaging Interface
+Each of the following functions must be thread safe.
+
+| Function Pointer | Description |
+| :-: | --- |
+| `MQTTAgentMessageSend_t` | A function that sends `MQTTAgentCommand_t *` pointers to be received by `MQTTAgent_CommandLoop`. This can be implemented by pushing to a thread safe queue. |
```

Comment:
Instead of providing the `MQTTAgentCommand_t *`, wouldn't it be easier for a reader to mention just generically that this function enables commands to be sent to a thread-safe queue. Similarly, for the `MQTTAgentMessageRecv_t` function, it can say that it enabled commands to be reach from a thread-safe queue

Reply:
I thought it's better to be more specific here since `MQTTAgentCommand_t *` is part of the function prototype, so they will be dealing with it when implementing the function.